### DISCUSSION TITLE
Any plans to add to HA as an official integration?

### DISCUSSION THREAD

**Discussion by shred86** [humbertogontijo/homeassistant-roborock#46] [General]
First off, thank you @humbertogontijo for making this integration! I was curious if you were planning on trying to add this as an official integration to HA or just keeping it as a 3rd party integration via HACS? I realize this is early in development and it’s a lot easier to develop with everything in the same library. However, if you were planning on making this an official integration, then a separate library for just the “API” would have to be created. If that’s the case, is it worth using aiohttp instead of requests for async support?

**Comment by humbertogontijo** [comment]
Hey @shred86. Thanks. I am not really aware to how to create an official integration. I'm ok with changing the library to aiohttp. I'll try to find documentation to how to request an official integration

**Reply by shred86** [reply]
I think the biggest change is separating the API from the integration. So the API would have to become its own library published in PyPi (e.g. “roborockpy”) and this integration would import that library. I recently did some work on another integration that was similar to this where we had to separate all the API specific code, and it just seemed like it would have been easier to design it that way from the beginning.

Same thing with the asyncio support - it’s probably easier to implement now and start coding everything async than go back and try to convert it later, but if the API specific code is going to be separated out into its own library, might be a good time to accomplish both at the same time.

Anyways, thanks again for this integration and looking forward to see the continued development! I’d be happy to help where I can, but I’m just a hobbyist when it comes to coding/python so probably won’t be of much use other than just reviewing and trying to help with issues. ;)

**Comment by Lash-L** [comment]
> Hey @shred86. Thanks. I am not really aware to how to create an official integration. I'm ok with changing the library to aiohttp. I'll try to find documentation to how to request an official integration

I am currently going through the process of adding my own direct integration to HA (just waiting for the integration to be PR approved), I'd be happy to help you do the same for Roborock. I really appreciate this integration and I am happy to help in any way I can.

**Reply by MarkkraMs** [reply]
Would greatly increase reach (and for HA novices like me limit the complexity a bit)

**Comment by Lash-L** [comment]
@humbertogontijo I'm happy to start working on this if you're okay with it. I'd add you to the repo and try to take care of getting it running. It wouldn't be too much work

**Comment by Lash-L** [comment]
For anyone who is interested, check this comment here: https://github.com/humbertogontijo/homeassistant-roborock/issues/11#issuecomment-1488682784

I currently have a PR for adding roborock to core, and there are a few ways you can help.

**Comment by peteh** [comment]
@Lash-L What's the plan for the future? Will the hacs integration coexist or will the official integration be the one that will be maintained with bugfixes? As it seems this is stable now I'm willing to switch from the xiaomi integration, but I don't know which integration (ha official or hacs) is the best to go forward for long term maintenance.

**Reply by Lash-L** [reply]
For now - the hacs integration has more features.

We will continue to develop in parallel and whenever I add functionality to core, I'll add it to the custom as well.

In the future the custom integration will probably get features first, but the core integration should always theoretically be more stable and shouldn't be too far behind in terms of features.

When they reach parity, I'm not sure if the core one will be the "primary" one or not. That would be up to @humbertogontijo

**Reply by humbertogontijo** [reply]
Hard to tell, but I'd assume that what @Lash-L said is correct. Using custom integration for beta versions and the core one for stable releases
